### PR TITLE
Remove wrapped Executes and Responses

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description
Now that every EA is in TS:
- Removes `ExecuteWrappedResponse`
- Removes `util.wrapExecute` from every EA
- Removes `withAsync`

### Issue
#177

### Ticket
[176639347](https://www.pivotaltracker.com/story/show/176639347)

### CODE REVIEW

File: bootstrap/src/index.ts [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-} from '@chainlink/types'
+import { ExecuteSync, AdapterRequest, Execute, AdapterHealthCheck } from '@chainlink/types'
 
-type Middleware = (execute: ExecuteWrappedResponse) => Promise<ExecuteWrappedResponse>
+type Middleware = (execute: Execute) => Execute
+type AsyncMiddleware = (execute: Execute) => Promise<Execute>
 
 // Try to initialize, pass through on error
-const skipOnError = (middleware: Middleware) => async (execute: ExecuteWrappedResponse) => {
+const skipOnError = (middleware: Middleware | AsyncMiddleware) => async (execute: Execute) => {
```
Comment (krebernisak) [inline bootstrap/src/index.ts:13]: I think we are good with having just one `Middleware` type, one that is async.

File: bootstrap/src/index.ts [hunk 2]
```diff
@@ -25,10 +20,12 @@ const skipOnError = (middleware: Middleware) => async (execute: ExecuteWrappedRe
 }
 
 // Make sure data has the same statusCode as the one we got as a result
-const withStatusCode = (execute: ExecuteWrappedResponse) => async (data_: AdapterRequest) => {
-  const { statusCode, data } = await execute(data_)
+// Is this still neccessary?
+const withStatusCode: Middleware = (execute) => async (data_: AdapterRequest) => {
+  const { statusCode, data, ...rest } = await execute(data_)
```
Comment (krebernisak) [inline bootstrap/src/index.ts:25]: What is the reason you introduced this extra `...rest` data to be outputted?

File: bootstrap/src/lib/cache/index.ts [hunk 3]
```diff
@@ -70,10 +70,9 @@ export const redactOptions = (options: CacheOptions) => ({
       : local.redactOptions(options.cacheOptions),
 })
 
-export const withCache = async (
-  execute: ExecuteWrappedResponse,
-  options: CacheOptions = defaultOptions(),
-) => {
+export const withCache = async (execute: Execute): Promise<Execute> => {
+  const options = defaultOptions()
```
Comment (krebernisak) [inline bootstrap/src/lib/cache/index.ts:74]: Why did this API change? Shouldn't anyone using the cache be able to configure the cache?

**Review by krebernisak — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -56,6 +56,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Removed
 
 - WTI Outlier detection and IV Outlier detection adapters are now covered in the Outlier detection composite adapter
+- Removes ExecuteWrappedResponse and WrappedAdapterResponse
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply CHANGELOG.md:59]: Yeah, good point.

File: bootstrap/src/index.ts (hunk 1 above)
Comment (justinkaseman) [reply bootstrap/src/index.ts:13]: Also a good point.

File: bootstrap/src/index.ts (hunk 2 above)
Comment (justinkaseman) [reply bootstrap/src/index.ts:25]: `AdapterResponse` has a required JobRunID
```
  export type AdapterResponse = {
    jobRunID: string
    statusCode: number
    data: any
    result?: any
  }
```

File: bootstrap/src/lib/cache/index.ts (hunk 3 above)
Comment (justinkaseman) [reply bootstrap/src/lib/cache/index.ts:74]: Yes they should. I was having a problem with the Middleware type, this shouldn't be changed. Reverted.

**Review by RodrigoAD — APPROVED** [review]

**Review by RodrigoAD — APPROVED** [review]

**Pull request merged**